hail/www/references.md

```diff
@@ -4,8 +4,26 @@ The following is an incomplete list of scientific work enabled by Hail. We welco
 
 In addition to software development, the Hail team engages in theoretical, algorithmic, and empirical research inspired by scientific collaboration. Examples include [Loss landscapes of regularized linear autoencoders](https://github.com/danielkunin/Regularized-Linear-Autoencoders), [Secure multi-party linear regression at plaintext speed](https://github.com/jbloom22/DASH), and [A synthetic-diploid benchmark for accurate variant-calling evaluation](https://www.nature.com/articles/s41592-018-0054-7).
 
+*Last updated on February 3rd, 2020 at 11:00 AM EST*
```

Comment:
the time is probably too granular, might indicate that this is automatically generated. Let's just fix this next time we change it.